public/app/features/alerting/unified/api/buildInfo.ts

```diff
@@ -118,19 +118,15 @@ export async function discoverDataSourceFeatures(dsSettings: {
     };
   }
 
-  if (buildInfoResponse.data.application === 'Grafana Mimir') {
-    // Mimir does return a version, but this is the mimir version, and not the prometheus version.
-    // We do not want to use mimir version numbers to detect feature support, we should instead ask the mimir team to add new flags.
-    return {
-      application: PromApplication.Mimir,
-      features: {
-        rulerApiEnabled: features?.ruler_config_api === 'true',
-      },
-    };
-  }
-
-  // @todo remove
-  throw Error('Unknown prometheus type!');
+  // We've checked everything else, it must be Mimir!
+  // Mimir does return a version, but this is the mimir version, and not the prometheus version.
+  // We do not want to use mimir version numbers to detect feature support, we should instead ask the mimir team to add new flags.
+  return {
+    application: PromApplication.Mimir,
+    features: {
+      rulerApiEnabled: features?.ruler_config_api === 'true',
+    },
+  };
```

Comment:
Won't be nice if we keep that check for "Mimir" just in case?

Reply:
Well the thanos folks got back to me, it looks like they might have another API endpoint for thanos buildinfo, so I think I'm gonna add that conditional back in, and check if it's thanos if it's not mimir